test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -103,381 +108,665 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
+				gomega.Expect(createdWorkload.Spec.PodSets).To(gomega.HaveLen(1))
+				gomega.Expect(createdWorkload.Spec.PodSets[0].Count).To(gomega.Equal(int32(2)))
+				gomega.Expect(createdWorkload.Spec.QueueName).To(gomega.Equal("test-queue"), "The Workload should have .spec.queueName set")
+
+				admission := testing.MakeAdmission(clusterQueue.Name, "120fa2c0").
+					Assignment(corev1.ResourceCPU, "default", "1").
+					AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
+					Obj()
+				ginkgo.By("checking that all pods in group are unsuspended when workload is admitted", func() {
+					gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+					util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+
+					util.ExpectPodUnsuspendedWithNodeSelectors(ctx, k8sClient, pod1LookupKey, map[string]string{"kubernetes.io/arch": "arm64"})
+					util.ExpectPodUnsuspendedWithNodeSelectors(ctx, k8sClient, pod2LookupKey, map[string]string{"kubernetes.io/arch": "arm64"})
+
+					gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+					gomega.Expect(createdWorkload.Status.Conditions).Should(gomega.BeComparableTo(
+						[]metav1.Condition{
+							{Type: kueue.WorkloadQuotaReserved, Status: metav1.ConditionTrue},
+							{Type: kueue.WorkloadAdmitted, Status: metav1.ConditionTrue},
+						},
+						wlConditionCmpOpts...,
+					))
 				})
 
-				ginkgo.By("fetch the job and verify it is suspended as the checks are not ready", func() {
-					gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
-						g.Expect(k8sClient.Get(ctx, *podLookupKey, createdPod)).To(gomega.Succeed())
-						return createdPod.Spec.SchedulingGates
-					}, util.Timeout, util.Interval).Should(
-						gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
-					)
+				ginkgo.By("set the pods as running", func() {
+					util.SetPodsPhase(ctx, k8sClient, corev1.PodRunning, pod1, pod2)
 				})
 
-				ginkgo.By("fetch the created workload", func() {
-					gomega.Eventually(func() error {
-						return k8sClient.Get(ctx, *wlLookupKey, createdWorkload)
-					}, util.Timeout, util.Interval).Should(gomega.Succeed())
+				createdPod1 := &corev1.Pod{}
+				createdPod2 := &corev1.Pod{}
+				ginkgo.By("checking that pod group is still running when workload is evicted", func() {
+					gomega.Expect(func() error {
+						w := createdWorkload.DeepCopy()
+						workload.SetEvictedCondition(w, "ByTest", "by test")
+						return workload.ApplyAdmissionStatus(ctx, k8sClient, w, false)
+					}()).Should(gomega.Succeed())
+
+					gomega.Consistently(func(g gomega.Gomega) {
+						g.Expect(k8sClient.Get(ctx, pod1LookupKey, createdPod1)).To(gomega.Succeed())
+						g.Expect(k8sClient.Get(ctx, pod2LookupKey, createdPod2)).To(gomega.Succeed())
+					}, util.ConsistentDuration, util.Interval).Should(gomega.Succeed())
```

Comment:
No need for this. But we might want to check that the Pods get a DeletionTimestamp.